spring-integration-core/src/main/java/org/springframework/integration/codec/CompositeCodec.java

```diff
@@ -41,9 +39,20 @@ public class CompositeCodec implements Codec {
 
 	public CompositeCodec(Map<Class<?>, Codec> delegates, Codec defaultCodec) {
 		this.defaultCodec = defaultCodec;
+		Assert.notEmpty(delegates, "delegates must not be empty");
 		this.delegates = new HashMap<Class<?>, Codec>(delegates);
 	}
 
+	/**
+	 * @since 7.0
+	 *
+	 * @deprecated This constructor is deprecated because the delegates map must contain at least one entry.
+	 * Creating a {@code CompositeCodec} without any delegates is not a valid use case, as there would be
+	 * no codecs to delegate to for encoding or decoding operations. Please use the constructor that accepts
+	 * a non-empty delegates map to ensure proper delegation behavior.
```

Comment:
JavaDocs must be neutral and come with an official language without any `please` and similar.
How about this:
```
* @deprecated since 7.0 in favor of {@link #CompositeCodec(Map, Codec)} with some delegates.
```
I don't think we should say more here  about "why?".